### PR TITLE
Harden count user logic on upgrade

### PR DESCRIPTION
**Pull request by MorrisJobke**
In addition to #6001 and #5695

### CODE REVIEW

File: lib/base.php [hunk 1]
```diff
@@ -341,10 +341,14 @@ private static function printUpgradePage() {
...
-				$stats = \OC::$server->getUserManager()->countUsers();
-				$totalUsers = array_sum($stats);
-				$tooBig = ($totalUsers > 50);
+				try {
+					// count users
+					$stats = \OC::$server->getUserManager()->countUsers();
+					$totalUsers = array_sum($stats);
+					$tooBig = ($totalUsers > 50);
+				} catch (\Doctrine\DBAL\Exception\TableNotFoundException $e) {
+					\OC::$server->getLogger()->logException($e, ['level' => \OCP\Util::INFO, 'app' => 'core');
```
Comment (LukasReschke) [inline lib/base.php]: :bomb: :bomb:

ParseError: syntax error, unexpected ')', expecting ']' in /drone/src/github.com/nextcloud/server/pull/6002/lib/base.php:350

**Review by LukasReschke — CHANGES_REQUESTED** [review]

**Review by LukasReschke — APPROVED** [review]

**Comment by nickvergessen** [comment]
I don't get what this is for. `countUsers` only touches the database backends, they should work?
Accounts table is not touched by this...

**Review by nickvergessen — APPROVED** [review]
... but sure, anything is better than an exception

**Comment by MorrisJobke** [comment]
> I don't get what this is for. countUsers only touches the database backends, they should work?
> Accounts table is not touched by this...

I just want to avoid, that upstream somehow renames this table and then our backend will not be able to count anymore. But we can also not run this code and fix it once it happens. It could also be the case, that a column is renamed or so.

**Comment by MorrisJobke** [comment]
Makes no sense at all -> closing

**Pull request closed without merging**